### PR TITLE
libopae: add support to upstream driver sysfs changes

### PR DESCRIPTION
**Pull request by anandaravuri**
Sysfs paths changes:
-Port error
-Port error clear
-FPGA  power
-FPGA Thermal
-FPGA Perf counters

### CODE REVIEW

File: libopae/plugins/xfpga/reconf.c [hunk 1]
```diff
@@ -189,28 +189,31 @@ STATIC fpga_result open_accel(fpga_handle handle, fpga_handle *accel)
...
-	uint64_t error                     = 0 ;
+	char sysfs_path[PATH_MAX] = {0};
+	fpga_result result        = FPGA_OK;
+	uint64_t error            = 0 ;
 
-	result = get_port_sysfs(handle, sysfs_path);
+	result = sysfs_get_port_error_path(handle, sysfs_path);
 	if (result != FPGA_OK) {
-		FPGA_ERR("Failed to get port syfs path");
+		FPGA_ERR(" Failed to get port errors path");
```
Comment (tswhison) [inline libopae/plugins/xfpga/reconf.c]: Need the space between " and Failed?

File: libopae/plugins/xfpga/reconf.c [hunk 2]
```diff
@@ -189,28 +189,31 @@ STATIC fpga_result open_accel(fpga_handle handle, fpga_handle *accel)
...
+	result = sysfs_read_u64(sysfs_path, &error);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Failed to read port errors");
+		return result;
+	}
+
+	result = sysfs_get_port_error_clear_path(handle, sysfs_path);
 	if (result != FPGA_OK) {
-		FPGA_ERR("Failed to get port errors");
+		FPGA_ERR(" Failed to get port errors clear path");
```
Comment (tswhison) [inline libopae/plugins/xfpga/reconf.c]: Need the space between " and Failed?

File: libopae/plugins/xfpga/reconf.c [hunk 3]
```diff
@@ -400,8 +404,8 @@ fpga_result __FPGA_API__ xfpga_fpgaReconfigureSlot(fpga_handle fpga,
 		}
 
 		// Set power threshold for integrated fpga.
-		if (hw_type == FPGA_HW_MCP) {
-
+		if (hw_type == FPGA_HW_MCP &&
+			!stat(FPGA_SYSFS_CLASS_PATH_INTERNAL, &st)) {
```
Comment (tswhison) [inline libopae/plugins/xfpga/reconf.c]: Isn't it an error if the file doesn't exist?

Comment (anandaravuri) [reply libopae/plugins/xfpga/reconf.c]: Upstream driver doesn't support  set threshold for MCP, only internal driver supports it.

enum fails if  path doesn't exists FPGA_SYSFS_CLASS_PATH_INTERNAL

File: libopae/plugins/xfpga/reconf.c (hunk 2 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/reconf.c]: fixd

File: libopae/plugins/xfpga/reconf.c (hunk 1 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/reconf.c]: fixed

File: testing/xfpga/test_sysfs_c.cpp [hunk 4]
```diff
@@ -748,8 +748,66 @@ TEST_P(sysfs_c_mock_p, fpga_sysfs_03) {
   EXPECT_EQ(result, FPGA_OK);
 }
 
+TEST_P(sysfs_c_mock_p, fpga_sysfs_04) {
```
Comment (ibarraza) [inline testing/xfpga/test_sysfs_c.cpp]: Can we add some test descriptions to these tests?

Comment (anandaravuri) [reply testing/xfpga/test_sysfs_c.cpp]: fixed

File: libopae/plugins/xfpga/sysfs_int.h [hunk 5]
```diff
@@ -35,6 +35,9 @@
 
 #define SYSFS_PATH_MAX 256
 
+#define FPGA_SYSFS_CLASS_PATH_INTERNAL "/sys/class/fpga"
+#define FPGA_SYSFS_CLASS_PATH_UPSDRV  "/sys/class/fpga_region"
```
Comment (r-rojo) [inline libopae/plugins/xfpga/sysfs_int.h]: can we rename these to use the suffixes:
_INTEL (instead of INTERNAL)
_DFL (instead of UPSDRV)

Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs_int.h]: fixed

File: libopae/plugins/xfpga/sysfs.c [hunk 6]
```diff
@@ -693,6 +708,110 @@ fpga_result sysfs_get_interface_id(fpga_token token, fpga_guid guid)
...
+	fpga_result res = FPGA_OK;
+	struct _fpga_token *_token = (struct _fpga_token *)token;
+	ASSERT_NOT_NULL(_token);
+
+	if (sysfs_temp == NULL) {
+		FPGA_ERR("Invalid input parameters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = cat_token_sysfs_path(sysfs_temp, token, SYSFS_FORMAT(sysfs_fme_temp_glob));
```
Comment (r-rojo) [inline libopae/plugins/xfpga/sysfs.c]: If we have a glob pattern here, should't we resolve it here?

File: libopae/plugins/xfpga/sysfs.c [hunk 7]
```diff
@@ -693,6 +708,110 @@ fpga_result sysfs_get_interface_id(fpga_token token, fpga_guid guid)
...
+	fpga_result res = FPGA_OK;
+	struct _fpga_token *_token = (struct _fpga_token *)token;
+	ASSERT_NOT_NULL(_token);
+
+	if (sysfs_perf == NULL) {
+		FPGA_ERR("Invalid input parameters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = cat_token_sysfs_path(sysfs_perf, token, SYSFS_FORMAT(sysfs_fme_perf_glob));
```
Comment (r-rojo) [inline libopae/plugins/xfpga/sysfs.c]: Shouldn't we resolve the glob pattern here?

File: libopae/plugins/xfpga/sysfs.c (hunk 6 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: Metrics API glab enums folder with search pattern power*_*
Upstream driver Power sysfs is different from internal driver 


[lab@PSERA2-NCF22 mybuild]$ ls -l /sys/class/fpga_region/region0/dfl-fme.0/hwmon/hwmon4/
total 0
lrwxrwxrwx 1 root root    0 Nov 12 15:46 device -> ../../../dfl-fme.0
-r--r--r-- 1 root root 4096 Nov 12 15:46 name
drwxr-xr-x 2 root root    0 Nov 12 15:46 power
-rw-r--r-- 1 root root 4096 Nov 12 15:46 power1_crit
-r--r--r-- 1 root root 4096 Nov 12 15:46 power1_crit_alarm
-r--r--r-- 1 root root 4096 Nov 12 15:46 power1_fpga_limit
-r--r--r-- 1 root root 4096 Nov 12 15:46 power1_input
-r--r--r-- 1 root root 4096 Nov 12 15:46 power1_ltr
-rw-r--r-- 1 root root 4096 Nov 12 15:46 power1_max
-r--r--r-- 1 root root 4096 Nov 12 15:46 power1_max_alarm
-r--r--r-- 1 root root 4096 Nov 12 15:46 power1_xeon_limit
lrwxrwxrwx 1 root root    0 Nov 12 15:46 subsystem -> ../../../../../../../../class/hwmon
-rw-r--r-- 1 root root 4096 Nov 12 15:46 uevent

File: libopae/plugins/xfpga/sysfs.c (hunk 7 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: Metrics API glab enums folder with search pattern power*_*
Upstream driver TEMP sysfs is different from internal driver 


[lab@PSERA2-NCF22 mybuild]$ ls -l /sys/class/fpga_region/region0/dfl-fme.0/hwmon/hwmon3/
total 0
lrwxrwxrwx 1 root root    0 Nov 12 15:49 device -> ../../../dfl-fme.0
-r--r--r-- 1 root root 4096 Nov 12 15:49 name
drwxr-xr-x 2 root root    0 Nov 12 15:49 power
lrwxrwxrwx 1 root root    0 Nov 12 15:49 subsystem -> ../../../../../../../../class/hwmon
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_crit
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_crit_alarm
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_emergency
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_input
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_max
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_max_alarm
-r--r--r-- 1 root root 4096 Nov 12 15:49 temp1_max_policy
-rw-r--r-- 1 root root 4096 Nov 12 15:46 uevent

**Pull request merged**